src/locktower.rs

```diff
@@ -0,0 +1,172 @@
+use std::collections::VecDeque;
+pub struct VoteLock {
+    /// Fork id, same as slot id. There can only be 1 fork per slot, since the empty PoH ticks
+    /// cannot be voted on.  A validator cannot vote on a fork unless that fork is connected to
+    /// a trunk. The starting trunk is the initial `finalized_fork`.
+    fork: u64,
+    /// Lockout is in terms of slots, so the slot height or fork id to unroll this
+    /// vote must be greater than fork + lockout.
+    lockout: u64,
+}
+
+pub enum LockTowerError {
+    VoteLocked,
+    VoteNotConverging,
+}
+
+pub struct LockTower {
+    vote_locks: VecDeque<VoteLock>,
+    /// The maximum size of vote_locks before reward is generated.  When the queue reached max_size,
+    /// a reward is generated and the oldest lock is removed.  That fork is considered finalized.
+    max_size: usize,
```

Comment:
Why not use `vote_locks.capacity()`?

Reply:
capacity is VecDeque implementation specific.  This value is reward specific. Seems dangerous to overload.  We can use a static sized array with a Deque interface.